tests/Testable/NavigationTests.elm

```diff
@@ -0,0 +1,245 @@
+                                )
+                , test "goes forward, asking to trigger location msg" <|
+                    \() ->
+                        init
+                            |> update (New "/foo")
+                            |> thenUpdate (New "/bar")
+                            |> thenUpdate (Jump -2)
+                            |> thenUpdate (Jump 1)
+                            |> Expect.equal
+                                ( ( 1
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    , getLocation "https://elm.testable/bar"
+                                    ]
+                                  )
+                                , TriggerLocationMsg (getLocation "https://elm.testable/foo")
+                                )
+                , test "has a limit to jumping back" <|
+                    \() ->
+                        init
+                            |> update (New "/foo")
+                            |> thenUpdate (Jump -100)
+                            |> Expect.equal
+                                ( ( 0
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    ]
+                                  )
+                                , TriggerLocationMsg initialLocation
+                                )
+                , test "has a limit to jumping forward" <|
+                    \() ->
+                        init
+                            |> update (New "/foo")
+                            |> thenUpdate (Jump -1)
+                            |> thenUpdate (Jump 100)
+                            |> Expect.equal
+                                ( ( 1
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    ]
+                                  )
+                                , TriggerLocationMsg (getLocation "https://elm.testable/foo")
+                                )
+                ]
+            , describe "visit url"
+                [ test "visit adds a new url to the history, asking to trigger location msg" <|
+                    \() ->
+                        init
+                            |> update (Visit "/foo")
+                            |> Expect.equal
+                                ( ( 1
+                                  , [ initialLocation
+                                    , getLocation "https://elm.testable/foo"
+                                    ]
+                                  )
+                                , TriggerLocationMsg (getLocation "https://elm.testable/foo")
+                                )
+                ]
+            ]
```

Comment:
I'm assuming all these tests were useful for while implementing.  I'm curious if you think they're still valuable to have around now that there are also tests in `NavigationTests` that I expect cover most if not all of the code tested here?  (I'm fine either way, but I'm curious what you think.)